doc/source/development/maintaining.rst

```diff
@@ -345,7 +345,126 @@ The benchmarks are scheduled by Airflow. It has a dashboard for viewing and debu
 Release process
 ---------------
 
-The process for releasing a new version of pandas can be found at https://github.com/pandas-dev/pandas-release
+The release process makes a snapshot of pandas (a git commit) available to users with
+a particular version number. After the release the new pandas version will be available
+in the next places:
+
+- Git repo with a new tag
+- Source distribution in a [GitHub release](https://github.com/pandas-dev/pandas/releases)
+- Pip packages in the [PyPI](https://pypi.org/project/pandas/)
+- Conda/Mamba packages in [conda-forge](https://anaconda.org/conda-forge/pandas)
+
+The process for releasing a new version of pandas is detailed next.
+
+The instructions contain ``<version>`` which needs to be replaced with the version
+to be released (e.g. ``1.5.2``). Also the branch to be released ``<branch>``, which
+depends on whether the version being released is the release candidate of a new version,
+or any other version. Release candidates are released from ``main``, while other
+versions are released from their branch (e.g. ``1.5.x``).
+
+
+Pre-release
+```````````
+
+1. Agree with the core team on the next topics:
+
+   - Release date
+   - Blockers (issues and PRs that must be part of the release)
+   - Next version after the one being released
+
+2. Update and clean release notes for the version to be released, including:
+
+   - Set the final date of the release
+   - Remove any unused bullet point
+   - Make sure there are no formatting issues, typos, etc.
+
+3. Make sure the CI is green for the last commit of the branch being released.
+
+4. If not a release candidate, make sure all backporting pull requests to the branch
+   being released are merged.
+
+5. Create a new issue and milestone for the version after the one being released.
```

Comment:
After a minor/major release, you would need to create two milestones, right?

One for the patch release (e.g. 1.5.1) and one for the next major version (e.g. 2.0.0).